realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -142,6 +155,39 @@ private SyncConfiguration(File directory,
         this.serverCertificateAssetName = serverCertificateAssetName;
         this.serverCertificateFilePath = serverCertificateFilePath;
         this.waitForInitialData = waitForInitialData;
+        this.forceSyncHistory = forceSyncHistory;
+    }
+
+    /**
+     * Returns a SyncConfiguration appropriate to open a synced Realm offline.
+     * This is useful when trying to open a backup Realm (after a Client Reset)
+     *
+     * @param canonicalPath the absolute path to the Realm file defined by this configuration.
+     * @param modules if specified it will restricts Realm schema to the provided module.
+     * @return SyncConfiguration that can be used offline
+     */
+    public static SyncConfiguration forOffline(String canonicalPath, Object... modules) {
```

Comment:
wouldn't it return a `RealmConfiguration` instead since it cannot be used for sync?

Reply:
Good point, I think Cocoa POC implementation used a [RLMRealmConfiguration](https://github.com/realm/realm-cocoa/pull/5259/files#diff-c64398ac8b77c37143815b54e83e9427R1121)